sdk/typespec/typespec_client_core/src/http/response.rs

```diff
@@ -45,193 +99,125 @@ impl RawResponse {
+    ///
+    /// Used to set the "type" of an untyped `Response<()>`, transforming it into a `Response<T>`.
+    pub fn map_body<T>(self) -> Response<T> {
+        Response {
+            status: self.status,
+            headers: self.headers,
+            body: self.body,
             phantom: PhantomData,
         }
     }
 }
 
-/// An HTTP response to deserialize a stream of bytes into a model of type `T`.
-pub struct Response<T> {
-    response: RawResponse,
-    phantom: PhantomData<T>,
-}
-
-impl<T> Response<T> {
-    /// Create an HTTP response.
-    pub fn new(status: StatusCode, headers: Headers, stream: PinnedStream) -> Self {
-        Self {
-            response: RawResponse::new(status, headers, stream),
-            phantom: PhantomData,
-        }
-    }
-
-    /// Get the status code from the response.
-    pub fn status(&self) -> StatusCode {
-        self.response.status
-    }
-
-    /// Get the headers from the response.
-    pub fn headers(&self) -> &Headers {
-        &self.response.headers
-    }
-
-    /// Deconstruct the HTTP response into its components.
-    pub fn deconstruct(self) -> (StatusCode, Headers, ResponseBody) {
-        (
-            self.response.status,
-            self.response.headers,
-            self.response.body,
-        )
-    }
-
-    /// Consume the HTTP response and return the HTTP body bytes.
-    pub fn into_body(self) -> ResponseBody {
-        self.response.body
-    }
-
-    /// Get the response body as the specified type from JSON.
-    pub async fn json(self) -> crate::Result<T>
-    where
-        T: DeserializeOwned,
-    {
-        self.into_body().json().await
-    }
-
-    /// Get the response body as the specified type from XML.
-    #[cfg(feature = "xml")]
-    pub async fn xml(self) -> crate::Result<T>
-    where
-        T: DeserializeOwned,
-    {
-        self.into_body().xml().await
+impl<T: Model> Response<T> {
+    /// Fetches the entire body and tries to convert it into type `T`.
+    ///
+    /// This is the preferred method for parsing the body of a service response into it's default model type.
+    ///
+    /// # Example
+    /// ```rust
+    /// # #[derive(serde::Deserialize)]
+    /// # pub struct GetSecretResponse {
+    /// #   name: String,
+    /// #   value: String,
+    /// # }
+    /// # typespec_client_core::json_model!(GetSecretResponse);
+    /// # pub struct SecretClient { }
+    /// # impl SecretClient {
+    /// #   pub async fn get_secret(&self) -> typespec_client_core::http::Response<GetSecretResponse> {
+    /// #    typespec_client_core::http::Response::from_bytes(
+    /// #      http_types::StatusCode::Ok,
+    /// #      typespec_client_core::http::headers::Headers::new(),
+    /// #      "{\"name\":\"database_password\",\"value\":\"hunter2\"}",
+    /// #    )
+    /// #  }
+    /// # }
+    /// # pub fn create_secret_client() -> SecretClient {
+    /// #   SecretClient { }
+    /// # }
+    ///
+    /// # tokio_test::block_on(async {
```

Comment:
Alternatively, instead of adding yet another dependency (though, it seems we already had it, but I want to reduce them as we go), what about just hiding an async main e.g.?

```rust
/// # #[tokio::main]
/// # async fn main() {
///    // clipped
/// # }
```